### ISSUE TITLE
Running generate locally

### ISSUE THREAD

**Issue by kokujin** [generate/generate#34]
Before opening an issue, please:

- read the [contributing guidelines](https://github.com/generate/generate/blob/master/.github/contributing.md)
- [search for existing duplicate or closed issues](https://github.com/generate/generate/issues?utf8=%E2%9C%93&q=is%3Aissue)
- Do not open issues to ask for implementation help or to ask "how to" questions here. Instead ask on [StackOverflow](stackoverflow.com) or one of the services listed on the [readme](https://github.com/generate/generate/blob/master/README.md#community)

For bug reports, please provide the following details:

- **version**: what version of generate you were using when you experienced the bug?
- **[reduced test case](https://css-tricks.com/reduced-test-cases/)**: the minimum amount of detail and code to reproduce the bug
- **error messages**: please paste any error reports into the issue or a gist

Please wrap all code and error messages in [markdown code fences](https://help.github.com/articles/creating-and-highlighting-code-blocks/).

**Comment by tunnckoCore** [comment]
yes, it will work. you should access it wit node_modules/.bin/gen as usual, i believe. but it not make sense, its not so usable that way. by this i mean its uncomfortable

**Comment by jonschlinkert** [comment]
If you experience an issue with this, consider it a bug. I'd like for this to be supported. thanks for the issue

**Comment by jonschlinkert** [comment]
> its not so usable that way. by this i mean its uncomfortable

I have an idea for this... we already support macros, like this:

```sh
$ gen --macro=foo bar baz qux
```

Which will allow you to run the `bar baz qux` generators  with the `$ gen foo` command; So when `$ gen foo` is given, the command is _replaced_ with `$ gen bar baz qux`.  

But we might be able to prepend or append macros to the current command. Which would allow you to do something like this:

```sh
$ gen --macro=local.prepend node_modules/.bin/gen
```

Then, you could just do this:

```sh
$ gen local bar baz qux
```

Which would expand to:

```sh
$ gen node_modules/.bin/gen bar baz qux
```

Thoughts?

**Comment by tunnckoCore** [comment]
Dont think so it's unnecessary, imho. maybe i can be some sort of detection like if task name contains `/` understand it as path ... hm. in anyway, if we have prepend macro and expand to that you shown, what next? i mean.. `$ gen node_modules ...` not looks good to me, in anyway it would have some sort of detection as i said.

unnecessary complexity, when i believe it works currently or at least it will be simple fix in the cli.

**Comment by jonschlinkert** [comment]
> $ gen node_modules ...

why would you do that?

**Comment by tunnckoCore** [comment]
You said it that way.

> Which would expand to:
> `$ gen node_modules/.bin/gen bar baz qux`

in which i dont see sense, or not sense, but how this would make it better

**Comment by jonschlinkert** [comment]
> Which would expand to:

sorry, I see what you mean. I thought of that macro idea and didn't really think it through enough before I wrote it out. Of course, you would still need a global generate to resolve the macro and use it. 

Actually though, you could easily run a locally-installed generate like any other node library. You would just do:

```js
// -- generator.js-- (or whatever you want)
var generate = require('generate');
var app = generate();

app.task('default', function(cb) {
  cb();
});

app.build('default', function(err) {
  if (err) return console.log(err);
});
```

And run: 

```sh
$ node generator
```

**Comment by tunnckoCore** [comment]
Yea exactly. Or as I mention in first comment `node_modules/.bin/gen task task2 task3`.
There are a few ways to do what @kokujin need, and we don't need to add some more complexity.
